src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { DispatchContext, StateContext } from '../TodosContext/TodosContext';
+import { ActionTypes } from '../../types/types';
+
+type Props = {
+  title: string;
+  id: number;
+};
+
+export const TodoItem: React.FC<Props> = ({ title, id }) => {
+  const dispatch = useContext(DispatchContext);
+  const { todos } = useContext(StateContext);
+
+  const [newTitle, setNewTitle] = useState(title);
+  const [editing, setEditing] = useState(false);
```

Comment:
Rename all boolean variables according [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)
```suggestion
  const [isEditing, setIsEditing] = useState(false);
```